src/app/api/verify-membership-payment/route.ts

```diff
@@ -0,0 +1,78 @@
+import { db } from '@/db';
+import { members } from '@/db/schema';
+import { redisClient } from '@/lib/redis';
+import { squareClient } from '@/lib/square';
+import { currentUser } from '@clerk/nextjs';
+import { eq } from 'drizzle-orm';
+import { redirect } from 'next/navigation';
+import { ApiError } from 'square';
+import { z } from 'zod';
+
+/**
+ * Verify that the membership payment has been made and update the database to reflect the payment
+ * status
+ *
+ * A redirect URL can be specified as well.
+ */
+export async function PUT(request: Request) {
+    let req;
+    try {
```

Comment:
not necessary. only zod is fine

Reply:
If the JSON is invalid or there isn't any JSON at all, then the async function will throw and you'll get a 500 error. Unless there is a way to handle this with zod?

Reply:
ig `z.min(1)` has already check the param's existence

Reply:
yeah but in my testing, it throws before it even reaches the zod validation...